server/repository/mongodb/service_commitments.py

```diff
@@ -0,0 +1,56 @@
+"""
+Module for handling MongoDB operations related to service commitments.
+Provides methods to insert and fetch service commitments from the database.
+"""
+
+from config.mongodb_config import get_db
+
+class MongoRepoCommitments:
+    """
+    A MongoDB repository for managing service commitments.
+    """
+    def __init__(self):
+        """
+        Initialize the repository by setting up the database and collection.
+        """
+        self.db = get_db()
+        self.collection = self.db.service_commitments
+
+    def insert_service_commitments(self, commitments):
+        """
+        Inserts multiple service commitments into the database.
+        Removes any existing _id fields to let MongoDB generate unique IDs.
+
+        Args:
+            commitments (list): A list of dictionaries 
+            representing the service commitments.
+
+        Returns:
+            list: A list of inserted IDs for the commitments.
+        """
+        # Create a new list with _id removed from each commitment
+        cleaned_commitments = []
+        for commitment in commitments:
+            commitment_copy = commitment.copy()
+            commitment_copy.pop("_id", None)  # Remove _id if it exists
```

Comment:
```suggestion
            commitment_copy.pop("commitment_id", None)  # Remove _id if it exists
```
Our ServiceCommitment class does not have an _id field, it has a commitment_id. Mongo repo will create an _id field for every new record. If you rename commitment_id to _id in ServiceCommitment, then you can keep your current code.

Reply:
This is still a problem